website/docs/r/eventgrid_event_subscription.html.markdown

```diff
@@ -0,0 +1,143 @@
+---
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_eventgrid_event_subscription"
+sidebar_current: "docs-azurerm-resource-messaging-eventgrid-event-subscription"
+description: |-
+  Manages an EventGrid Event Subscription
+
+---
+
+# azurerm_eventgrid_event_subscription
+
+Manages an EventGrid Event Subscription
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "default" {
+  name     = "defaultResourceGroup"
+  location = "West US 2"
+}
+
+resource "azurerm_storage_account" "default" {
+	name                     = "defaultStorageAccount"
+	resource_group_name      = "${azurerm_resource_group.default.name}"
+	location                 = "${azurerm_resource_group.default.location}"
+	account_tier             = "Standard"
+	account_replication_type = "LRS"
+  
+	tags {
+	  environment = "staging"
+	}
+  }
+  
+  resource "azurerm_storage_queue" "default" {
+	name                 = defaultStorageQueue"
+	resource_group_name  = "${azurerm_resource_group.default.name}"
+	storage_account_name = "${azurerm_storage_account.default.name}"
+  }
+
+resource "azurerm_eventgrid_event_subscription" "default" {
+  name                = "defaultEventSubscription"
+  scope = "${azurerm_resource_group.default.id}"
+  storage_queue_endpoint {
+	  storage_account_id = "${azurerm_storage_account.default.id}"
+	  queue_name = "${azurerm_storage_queue.default.name}"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name of the EventGrid Event Subscription resource. Changing this forces a new resource to be created.
+
+* `scope` - (Required) Specifies the scope at which the EventGrid Event Subscription should be created. Changing this forces a new resource to be created.
+
+* `event_delivery_schema` - (Optional) Specifies the event delivery schema for the event subscription. Possible values include: `EventGridSchema`, `CloudEventV01Schema`, `CustomInputSchema`.
+
+* `topic_name` - (Optional) Specifies the name of the topic to associate with the event subscription.
+
+* `storage_queue_endpoint` - (Optional) A `storage_queue_endpoint` block as defined below.
+
+* `eventhub_endpoint` - (Optional) A `eventhub_endpoint` block as defined below.
+
+* `hybrid_connection_endpoint` - (Optional) A `hybrid_connection_endpoint` block as defined below.
+
+* `webhook_endpoint` - (Optional) A `webhook_endpoint` block as defined below.
+
```

Comment:
Could we add a note letting readers know that one of the above blocks is required?